File: backend/LogMin.Infrastructure/Persistence/Entities/Log.cs
namespace LogMin.Infrastructure.Persistence.Entities;

public class Log
{
    public Guid Id { get; set; }

    public string Message { get; set; } = default!;

    public string? StackTrace { get; set; }

    public string ServiceName { get; set; } = default!;

    public DateTime Timestamp { get; set; }

    public bool Processed { get; set; }

    public DateTime? ProcessedAt { get; set; }

    public string? Pattern { get; set; }

    public double? Score { get; set; }

    public Guid? IssueId { get; set; }
    public Issue? Issue { get; set; }
}

File: frontend/logmind-dashboard/src/app/core/models/log.ts
export interface LogDto {
  id: string;
  message: string;
  stackTrace?: string | null;
  serviceName: string;
  timestamp: string;
  processed: boolean;
  processedAt?: string | null;
  pattern?: string | null;
  score?: number | null;
  issueId?: string | null;
}

export interface LogsQuery {
  serviceName?: string;
  pattern?: string;
  issueId?: string;
  from?: string;
  to?: string;
  page?: number;
  pageSize?: number;
}
